modules/components/org.wso2.carbon.appfactory.application.mgt/src/main/java/org/wso2/carbon/appfactory/application/mgt/listners/InitialArtifactDeployerHandler.java

```diff
@@ -145,7 +146,7 @@ public void onCreation(Application application, String userName, String tenantDo
 			List<Container> containerList = new ArrayList<>();
 			containerList.add(container);
 			DeploymentConfig deploymentConfig = new DeploymentConfig();
-			String deploymentName = application.getName() + "-" + stage;
+			String deploymentName = application.getId() + "-" + stage;
```

Comment:
This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/b54500739fe65b9908eda6176ecbace4b76a972d